cmd/kube-rbac-proxy/app/sanitazion.go

```diff
@@ -0,0 +1,70 @@
+package app
+
+import (
+	"bytes"
+	"encoding/json"
+)
+
+// SanitizingFilter implements the LogFilter interface from klog with custom functions to detect and mask tokens.
+type SanitizingFilter struct{}
+
+// Filter is the filter function for non-formatting logging functions of klog.
+func (sf *SanitizingFilter) Filter(args []interface{}) []interface{} {
+	for i, v := range args {
+		if strValue, ok := v.(string); ok {
+			if containsTokenReview(strValue) {
+				args[i] = maskTokenInLog(strValue)
+			}
+		}
+	}
+	return args
+}
+
+// FilterF is the filter function for formatting logging functions of klog.
+func (sf *SanitizingFilter) FilterF(format string, args []interface{}) (string, []interface{}) {
+	for i, v := range args {
+		if strValue, ok := v.(string); ok {
```

Comment:
Golang encourages us to view at this as `[]byte`. How would the code change, if we would:

```suggestion
		if strValue, ok := v.([]byte); ok {
```

Reply:
@ibihim I agree with you, but in this case all of our logs are stored and processed as `string`. converting them to `[]byte` will add complexity as we'd have to convert them back to `string` for log output anyway. wdyt?

Reply:
Sure, then lets change the other functions.